doc/developer/path-internals-daemon.rst

```diff
@@ -0,0 +1,111 @@
+PATHD Internals
+===============
+
+Architecture
+------------
+
+Overview
+........
+
+The pathd deamon manage the segment routing policies, it owns the data
+structures representing them and can load modules that manipulate them like the
+PCEP module. Its responsibility is to select a candidate path for each configured
+policies and installed it in Zebra.
+
+Zebra
+.....
+
+Zebra manage policies that are active or pending to be activated due to the next
+hop not being available yet. In zebra, policies structure and API is defined in
+`zebra_srte.[hc]`.
+
+The responsibilities of Zebra are:
+
+ - Store the policies segment list.
+ - Install the policies when their next-hop is available.
+ - Notify other daemons of the status of the policies.
+
+Adding and removing policies is done using the commands `ZEBRA_SR_POLICY_SET`
+and `ZEBRA_SR_POLICY_DELETE` as parameter of the function `zebra_send_sr_policy`
+all defined in `zclient.[hc]`.
+
+If the first segment of the policy is an unknown label, it is kept until
+notified by the mpls hooks `zebra_mpls_label_created`, and then it is installed.
+
+To get notified when a policy status changes, a client can implement the
+`sr_policy_notify_status` callback defined in `zclient.[hc]`.
+
+For encoding/decoding the various data structures used to comunicate with zebra,
+the following functions are available from `zclien.[hc]`:
+`zapi_sr_policy_encode`, `zapi_sr_policy_decode` and
+`zapi_sr_policy_notify_status_decode`.
+
+
+Pathd
+.....
+
+
+Pathd daemon manages all the possible candidate paths for the segment routing
+policies and selects the best one following the configured rules. It also
```

Comment:
hmm those rules are actually not configured, they are specified by an RFC :D

you mean "... following the configuration and other data." maybe?